### PR TITLE
Improve documentation (0.9.10)

### PR DESCRIPTION
**Pull request by screamerbg**
* Add documentation for publishing local/new repositories #362 
* Add documentation regarding the specific Python, Git and Mercurial versions that mbed CLI supports #376

Reviewers:
- [x] @iriark01 
- [x] @sg-

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -691,6 +691,22 @@ As with any mbed CLI command, `mbed update` uses the current directory as a work
 
 <span class="tips">**Tip: Synchronizing library references:** Before triggering an update, you might want to synchronize any changes that you've made to the program structure by running ``mbed sync``, which will update the necessary library references and get rid of the invalid ones.</span>
 
+## Publishing local program or library
```
Comment (iriark01) [inline README.md]: ## Publishing a local program or library

File: README.md [hunk 2]
```diff
@@ -691,6 +691,22 @@ As with any mbed CLI command, `mbed update` uses the current directory as a work
...
+
+When you create new (local) source control managed program or library, all revision history will exist only locally and the repository wouldn't be associated with remote one. To publish the local repository (while preserving the revision history) please follow these steps:
```
Comment (iriark01) [inline README.md]: When you create a new (local) source-control managed program or library, its revision history exists only locally; the repository is not associated with the remote one. To publish the local repository without losing its revision history, please follow these steps:

File: README.md [hunk 3]
```diff
@@ -691,6 +691,22 @@ As with any mbed CLI command, `mbed update` uses the current directory as a work
...
+1. Create new empty repository on the remote side - either based on public repository hosting service (Github, Bitbucket, mbed.org), service of your own or even different location on your system
+2. Copy the URL/location of the new repository in your clipboard and open command line in the local repository directory (for example change directory to `mbed-os-example/local-lib`).
+3. To associate the local repository:
+ * For Git - run `git remote add origin <url-or-paht-to-your-remote-repo>`
+ * For Mercurial - edit .hg/hgrc and add (or replace if exists):
+ ```
+[paths]
+default = <url-or-paht-to-your-remote-repo>
+```
+4. Run `mbed publish` to publish your changes
```
Comment (iriark01) [inline README.md]: Please add a period to the end of steps. It's not a universal law but it's the ARM style

File: README.md [hunk 4]
```diff
@@ -691,6 +691,22 @@ As with any mbed CLI command, `mbed update` uses the current directory as a work
...
+
+In scanario with nested local repositories, start with the leaf repositories first.
```
Comment (iriark01) [inline README.md]: "In a scenario"
"start with the leaf repositories" (you don't need "first" - you already said "start with")

File: README.md [hunk 5]
```diff
@@ -54,11 +54,11 @@ mbed CLI is supported on Windows, Linux and Mac OS X. We're keen to learn about
 
 ### Requirements
 
-* **Python** - mbed CLI is a Python script, so you'll need Python in order to use it. mbed CLI was tested with [version 2.7.9 of Python](https://www.python.org/download/releases/2.7/).
+* **Python** - mbed CLI is a Python script, so you'll need Python in order to use it. mbed CLI was tested with [version 2.7.9 of Python](https://www.python.org/download/releases/2.7/) and is not compatible with Python 3 at the moment of this writing.
```
Comment (iriark01) [inline README.md]: You don't need "of this writing"

File: README.md [hunk 6]
```diff
@@ -54,11 +54,11 @@ mbed CLI is supported on Windows, Linux and Mac OS X. We're keen to learn about
...
 
 * **Git and Mercurial** - mbed CLI supports both Git and Mercurial repositories, so you'll need to install both:
-    * [Git](https://git-scm.com/).
-    * [Mercurial](https://www.mercurial-scm.org/).
+    * [Git](https://git-scm.com/) - version 1.9.5 or later
+    * [Mercurial](https://www.mercurial-scm.org/) - version or 2.2.2 or later
```
Comment (iriark01) [inline README.md]: "version 2.2.2 or later" (I assume that other "or" is there by mistake)

**Comment by sg-** [comment]
LGTM 👍

**Pull request merged**